src/Extension/FluentExtension.php

```diff
@@ -430,7 +430,7 @@ public function updateDeleteTables(&$queriedTables)
 
             // If table isn't localised, simple delete
             if (!isset($localisedTables[$table])) {
-                $rootDelete->execute();
+                $rootDelete->addWhere(['"ID"' => $this->owner->ID])->execute();
```

Comment:
Can you move the addWhere() up to outside of the loop? We use `$rootDelete` further down, and that also needs the ID filter. Please qualify the ID column also with the table name. :)

Reply:
```php
$rootDelete = SQLDelete::create("\"{$rootTable}\"")
	->addWhere(["\"{$rootTable}\".\"ID\"" => $this->owner->ID]);
```

Reply:
If possible, could you also please write a test for this fix?
That is, at test that deleting a non-localised record works as expected (as opposed to removing _all_ records)?

Reply:
It's a bit of a problem actually... you need to test the record is deleted at the right time. If you delete this row, you need to delete the parent row too. However, THAT table could still be needed for other locales, so you can't just delete it then either.

The real fix is a pain. I'll write up something tomorrow.

Reply:
Test:
 - create record in Locale A
 - create version in locale B
 - delete in locale B - base records exist, but _localised rows deleted
 - delete in locale A - both base and _localised records deleted (since it's the last locale).

Reply:
Thanks, guys. For now, I just updated the commit to lift the fix out of the loop as @tractorcow suggested.